File: Recusrion/python/cnt_of_subsequence_with_k_sum.py
def rec(idx , arr: list[int], sum , tot, n):
    
    if sum > tot:
        return 0
    if idx>=n:
        if sum == tot:
            return 1
        return 0
    
    sum+=arr[idx]
    l = rec(idx+1, arr, sum, tot, n)
    sum-=arr[idx]
    r = rec(idx+1, arr, sum, tot, n)
    
    return l + r


if __name__ == "__main__":
    
    a = [1,2,1]
    tot=2
    n = 3
    print(rec(0,a,0,tot,n))
    

File: Recusrion/c++/cnt_of_subsequence_with_k_sum.cpp
#include<bits/stdc++.h>
using namespace std;

int  rec(int idx ,int sum , int tot , int n , int arr[])
{
    if (sum > tot)
    {
        return 0;
    }
    if(idx >= n)
    {
        if(sum == tot)
        {
            return 1;
        }
        return 0;
    }
    sum+=arr[idx];
    int l =rec(idx+1,sum,tot,n,arr);

    sum-=arr[idx];
    int r=rec(idx+1,sum,tot,n,arr);

    return l+r;


}
int main()
{  
    int n = 4;
    int a[] = {1,2,1,0};
    int tot = 2;
    cout<<rec(0,0,tot,n,a);
    
    return 0;

}